File: backend/aci/common/schemas/quota.py
from pydantic import BaseModel

from aci.common.schemas.plans import PlanFeatures


class PlanInfo(BaseModel):
    """plan info"""

    name: str
    features: PlanFeatures


class QuotaUsageResponse(BaseModel):
    """complete quota usage response"""

    projects_used: int
    linked_accounts_used: int  # unique account owner IDs across org
    agent_credentials_used: int  # total org secrets in secrets table
    plan: PlanInfo

File: frontend/src/lib/types/quota.ts
export interface QuotaUsage {
  projects_used: number;
  linked_accounts_used: number;
  agent_credentials_used: number;
  plan: {
    name: string;
    features: {
      projects: number;
      linked_accounts: number;
      api_calls_monthly: number;
      agent_credentials: number;
      developer_seats: number;
      custom_oauth: boolean;
      log_retention_days: number;
    };
  };
}
